app/models/ability.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    def initialize(user)
+      user ||= User.new # Guest user (not logged in)
+  
+      # Define abilities for different roles
+      if user.admin?
+        can :manage, :all # Admins can manage all resources
+      else
+        can :read, :all
+        can :create, Post
+        can :destroy, Post, author_id: user.id
+        can :destroy, Comment, user_id: user.id
+  
+      end
+    end
+end
```

Comment:
- Good job done so far, but there is a syntax error in this file that needs to be fixed, You should have just one `def initialize (user)` method in your file. But you have two of them( One has no `end` ), which causes an error, please fix this 🙏.  

```ruby 
class Ability
    include CanCan::Ability
    def initialize(user)
      user ||= User.new # Guest user (not logged in)

      # Define abilities for different roles
      if user.admin?
        can :manage, :all # Admins can manage all resources
      else
        can :read, :all
        can :create, Post
        can :destroy, Post, author_id: user.id
        can :destroy, Comment, user_id: user.id

      end
    end
end
```